### PR TITLE
tests: Fix test failures for Local storage and Basic zones

### PR DESCRIPTION
**Pull request by davidjumani**
### Description

This PR skips / fixes tests in Local storage and SG enabled zones

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?

TODO

### CODE REVIEW

File: test/integration/smoke/test_direct_download.py [hunk 1]
```diff
@@ -184,20 +184,28 @@ def setUpClass(cls):
...
+                cls.network_offering.update(cls.apiclient, state='Enabled')
+                cls.services["network"]["networkoffering"] = cls.network_offering.id
+                cls.network = Network.create(
+                    cls.apiclient,
+                    cls.services["l2-network"],
+                    zoneid=cls.zone.id,
+                    networkofferingid=cls.network_offering.id
+                )
+                cls._cleanup.append(cls.network)
+                cls._cleanup.append(cls.network_offering)
```
Comment (DaanHoogland) [inline test/integration/smoke/test_direct_download.py:208]: these should be reversed and nearer to the respective creations
```suggestion
                cls.network_offering = NetworkOffering.create(
                    cls.apiclient,
                    cls.services["l2-network_offering"],
                )
                cls._cleanup.append(cls.network_offering)
                cls.network_offering.update(cls.apiclient, state='Enabled')
                cls.services["network"]["networkoffering"] = cls.network_offering.id
                cls.network = Network.create(
                    cls.apiclient,
                    cls.services["l2-network"],
                    zoneid=cls.zone.id,
                    networkofferingid=cls.network_offering.id
                )
                cls._cleanup.append(cls.network)
```

and also the tearDownClass should read:

```
    @classmethod
    def tearDownClass(cls):
        super(TestDirectDownloadTemplates, cls).tearDownClass()
```

(could be a separate PR but I tend to touch no test files without these kind of changes)

File: test/integration/smoke/test_direct_download.py [hunk 2]
```diff
@@ -269,6 +277,21 @@ def createServiceOffering(self, name, type, tags):
...
+                self.services["virtual_machine"],
+                serviceofferingid=offering.id
+            )
+        else :
+            vm = VirtualMachine.create(
+                self.apiclient,
+                self.services["virtual_machine"],
+                serviceofferingid=offering.id,
+                networkids=self.network.id
+            )
```
Comment (DaanHoogland) [inline test/integration/smoke/test_direct_download.py:293]:
```suggestion
            )
            self.cleanup.append(vm)
```

File: test/integration/smoke/test_direct_download.py [hunk 3]
```diff
@@ -269,6 +277,21 @@ def createServiceOffering(self, name, type, tags):
             tags=tags
         )
 
+    def deployVM(self, offering) :
+        if self.zone.networktype == 'Basic' :
+            vm = VirtualMachine.create(
+                self.apiclient,
+                self.services["virtual_machine"],
+                serviceofferingid=offering.id
+            )
```
Comment (DaanHoogland) [inline test/integration/smoke/test_direct_download.py:286]: this should add
```suggestion
            vm = VirtualMachine.create(
                self.apiclient,
                self.services["virtual_machine"],
                serviceofferingid=offering.id
            )
            self.cleanup.append(vm)
```
and not outside the method.
Same for createServiceOffering

**Review by DaanHoogland — APPROVED** [review]
generally good but some comments on further improvements

File: test/integration/smoke/test_direct_download.py (hunk 1 above)
Comment (DaanHoogland) [reply test/integration/smoke/test_direct_download.py:208]: and btw 
```
    def tearDown(self):
        super(TestDirectDownloadTemplates, swlf).tearDown()
```

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@davidjumani is tests for basic run using advanced tags parameter while evoking nosetests?

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test

**Review by rhtyd — APPROVED** [review]

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test

**Comment by sureshanaparti** [comment]
merging this based on approvals and smoke tests
